src/dlload.c

```diff
@@ -63,10 +63,14 @@ DLLEXPORT int jl_uv_dlopen(const char *filename, jl_uv_libhandle lib_, unsigned
 #endif
                          );
     if (lib->handle) {
+        if (lib->errmsg)
+            free(lib->errmsg);
```

Comment:
OK, turns out there is a code path in `BinDeps` that calls `jl_uv_dlopen` directly without setting `errmsg` to `NULL`, which is the culprit behind #10829.